cmake/introspection.cmake

```diff
@@ -231,3 +231,25 @@ int main()
   sysctl(name, 2, nullptr, nullptr, nullptr, 0);
 }
 " HAVE_SYSCTL_ARND)
+
+check_cxx_source_compiles("
+#include <stdint.h>
+#include <type_traits>
+
+int main()
+{
+  static_assert(std::is_same<int8_t, char>::value);
+}
+" CHAR_EQUALS_INT8)
+
+check_cxx_source_compiles("
+int foo(void) __attribute__((visibility(\"default\")));
+int main(){}
+" HAVE_DEFAULT_VISIBILITY_ATTRIBUTE)
+
+if(WIN32)
```

Comment:
Again, no need to special-case this to Win32. One day `__declspec(dllexport)` may work for FreeBSD, who knows? :)

Reply:
[Updated](https://github.com/hebasto/bitcoin/pull/6#issuecomment-1435777703).

Reply:
> One day `__declspec(dllexport)` may work for FreeBSD

:open_mouth: :angry: :fire: :ice_cube: :no_entry: ~!*@&^(_ !!!